Update documentation changelog for Godot 3.3+ (#4861)
godotengine/godot-docs: about/docs_changelog.rst

=======================

> The documentation is continually being improved. The release of version 3.2

Reviewer: Remove references to 3.2 and 3.1 in this paragraph maybe?

> includes many new pages, many fixes and updates for old pages, and many updates

Reviewer: ```suggestion
includes many new pages, many fixes and updates to existing pages, and many updates
```

to the :ref:`class reference <toc-class-ref>`. Below is a list of new pages
added since version 3.1.

.. note:: This document only contains new pages so not all changes are reflected,
          many pages have been substantially updated but are not reflected in this document.

New pages since version 3.3
---------------------------

> None yet.

Reviewer: Maybe we should list the pages added after 3.2.3 (or maybe after 3.2-stable) in the `3.2` branch as part of 3.3?

i.e. this would reflect what's new in 3.3 docs for users who last checked them when 3.2 was out.

Reviewer: Using the history of https://github.com/godotengine/godot-docs-l10n/blob/master/templates_list.txt we can know which page was added around which time

That's the diff since ~3.2 (March 2020, previous sync was in December 2019 so before 3.2 release):
```diff
diff --git a/templates_list.txt b/templates_list.txt
index eba19a6ed..f3ae002b9 100644
--- a/templates_list.txt
+++ b/templates_list.txt
@@ -2,7 +2,10 @@ index.pot
 about/index.pot
 about/introduction.pot
 about/faq.pot
+about/troubleshooting.pot
+about/list_of_features.pot
 about/docs_changelog.pot
+about/release_policy.pot
 getting_started/step_by_step/index.pot
 getting_started/step_by_step/intro_to_the_editor_interface.pot
 getting_started/step_by_step/scenes_and_nodes.pot
@@ -25,9 +28,11 @@ getting_started/step_by_step/filesystem.pot
 getting_started/step_by_step/scene_tree.pot
 getting_started/step_by_step/singletons_autoload.pot
 getting_started/editor/index.pot
-getting_started/editor/unity_to_godot.pot
+getting_started/editor/using_the_web_editor.pot
 getting_started/editor/command_line_tutorial.pot
 getting_started/editor/external_editor.pot
+getting_started/editor/default_key_mapping.pot
+getting_started/editor/unity_to_godot.pot
 getting_started/scripting/index.pot
 getting_started/scripting/gdscript/index.pot
 getting_started/scripting/gdscript/gdscript_basics.pot
@@ -35,20 +40,30 @@ getting_started/scripting/gdscript/gdscript_advanced.pot
 getting_started/scripting/gdscript/gdscript_exports.pot
 getting_started/scripting/gdscript/gdscript_styleguide.pot
 getting_started/scripting/gdscript/static_typing.pot
+getting_started/scripting/gdscript/warning_system.pot
 getting_started/scripting/gdscript/gdscript_format_string.pot
 getting_started/scripting/visual_script/index.pot
 getting_started/scripting/visual_script/what_is_visual_scripting.pot
 getting_started/scripting/visual_script/getting_started.pot
 getting_started/scripting/visual_script/nodes_purposes.pot
+getting_started/scripting/visual_script/custom_visualscript_nodes.pot
 getting_started/scripting/c_sharp/index.pot
 getting_started/scripting/c_sharp/c_sharp_basics.pot
 getting_started/scripting/c_sharp/c_sharp_features.pot
 getting_started/scripting/c_sharp/c_sharp_differences.pot
 getting_started/scripting/c_sharp/c_sharp_style_guide.pot
 getting_started/scripting/cross_language_scripting.pot
+getting_started/scripting/creating_script_templates.pot
 getting_started/workflow/index.pot
 getting_started/workflow/project_setup/index.pot
 getting_started/workflow/project_setup/project_organization.pot
+getting_started/workflow/project_setup/version_control_systems.pot
+getting_started/workflow/assets/index.pot
+getting_started/workflow/assets/import_process.pot
+getting_started/workflow/assets/importing_images.pot
+getting_started/workflow/assets/importing_audio_samples.pot
+getting_started/workflow/assets/importing_translations.pot
+getting_started/workflow/assets/importing_scenes.pot
 getting_started/workflow/assets/escn_exporter/index.pot
 getting_started/workflow/assets/escn_exporter/material.pot
 getting_started/workflow/assets/escn_exporter/physics.pot
@@ -56,24 +71,19 @@ getting_started/workflow/assets/escn_exporter/lights.pot
 getting_started/workflow/assets/escn_exporter/mesh.pot
 getting_started/workflow/assets/escn_exporter/skeleton.pot
 getting_started/workflow/assets/escn_exporter/animation.pot
-getting_started/workflow/assets/index.pot
-getting_started/workflow/assets/import_process.pot
-getting_started/workflow/assets/importing_images.pot
-getting_started/workflow/assets/importing_audio_samples.pot
-getting_started/workflow/assets/importing_translations.pot
-getting_started/workflow/assets/importing_scenes.pot
 getting_started/workflow/export/index.pot
 getting_started/workflow/export/exporting_projects.pot
 getting_started/workflow/export/exporting_pcks.pot
 getting_started/workflow/export/feature_tags.pot
 getting_started/workflow/export/exporting_for_pc.pot
-getting_started/workflow/export/exporting_for_ios.pot
+getting_started/workflow/export/changing_application_icon_for_windows.pot
 getting_started/workflow/export/exporting_for_uwp.pot
+getting_started/workflow/export/exporting_for_ios.pot
 getting_started/workflow/export/exporting_for_android.pot
+getting_started/workflow/export/android_custom_build.pot
 getting_started/workflow/export/exporting_for_web.pot
+getting_started/workflow/export/exporting_for_dedicated_servers.pot
 getting_started/workflow/export/one-click_deploy.pot
-getting_started/workflow/export/android_custom_build.pot
-getting_started/workflow/export/changing_application_icon_for_windows.pot
 getting_started/workflow/best_practices/index.pot
 getting_started/workflow/best_practices/introduction_best_practices.pot
 getting_started/workflow/best_practices/what_are_godot_classes.pot
@@ -98,7 +108,7 @@ tutorials/2d/2d_sprite_animation.pot
 tutorials/3d/index.pot
 tutorials/3d/introduction_to_3d.pot
 tutorials/3d/using_transforms.pot
-tutorials/3d/3d_performance_and_limitations.pot
+tutorials/3d/3d_rendering_limitations.pot
 tutorials/3d/spatial_material.pot
 tutorials/3d/lights_and_shadows.pot
 tutorials/3d/reflection_probes.pot
@@ -133,14 +143,16 @@ tutorials/physics/ray-casting.pot
 tutorials/physics/ragdoll_system.pot
 tutorials/physics/kinematic_character_2d.pot
 tutorials/physics/soft_body.pot
+tutorials/physics/collision_shapes_3d.pot
 tutorials/math/index.pot
 tutorials/math/vector_math.pot
 tutorials/math/vectors_advanced.pot
 tutorials/math/matrices_and_transforms.pot
 tutorials/math/interpolation.pot
 tutorials/math/beziers_and_curves.pot
+tutorials/math/random_number_generation.pot
 tutorials/animation/index.pot
-tutorials/animation/introduction_2d.pot
+tutorials/animation/introduction.pot
 tutorials/animation/cutout_animation.pot
 tutorials/animation/2d_skeletons.pot
 tutorials/animation/animation_tree.pot
@@ -153,7 +165,6 @@ tutorials/io/index.pot
 tutorials/io/background_loading.pot
 tutorials/io/data_paths.pot
 tutorials/io/saving_games.pot
-tutorials/io/encrypting_save_games.pot
 tutorials/i18n/index.pot
 tutorials/i18n/internationalizing_games.pot
 tutorials/i18n/localization_using_gettext.pot
@@ -186,6 +197,7 @@ tutorials/shading/visual_shaders.pot
 tutorials/shading/intro_to_shaders_water_workshop.pot
 tutorials/shading/screen-reading_shaders.pot
 tutorials/shading/migrating_to_godot_shader_language.pot
+tutorials/shading/godot_shader_language_style_guide.pot
 tutorials/shading/advanced_postprocessing.pot
 tutorials/networking/index.pot
 tutorials/networking/high_level_multiplayer.pot
@@ -203,6 +215,7 @@ tutorials/vr/vr_primer.pot
 tutorials/vr/vr_starter_tutorial/index.pot
 tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_one.pot
 tutorials/vr/vr_starter_tutorial/vr_starter_tutorial_part_two.pot
+tutorials/vr/developing_for_oculus_quest.pot
 tutorials/plugins/index.pot
 tutorials/plugins/editor/index.pot
 tutorials/plugins/editor/installing_plugins.pot
@@ -219,7 +232,11 @@ tutorials/plugins/android/index.pot
 tutorials/plugins/android/android_plugin.pot
 tutorials/platform/index.pot
 tutorials/platform/android_in_app_purchases.pot
-tutorials/platform/services_for_ios.pot
+tutorials/platform/ios/index.pot
+tutorials/platform/ios/ios_plugin.pot
+tutorials/platform/ios/plugins_for_ios.pot
+tutorials/platform/platform_html5.pot
+tutorials/platform/html5_shell_classref.pot
 tutorials/platform/customizing_html5_shell.pot
 tutorials/platform/consoles.pot
 tutorials/threads/index.pot
@@ -233,8 +250,13 @@ tutorials/content/procedural_geometry/surfacetool.pot
 tutorials/content/procedural_geometry/immediategeometry.pot
 tutorials/content/making_trees.pot
 tutorials/optimization/index.pot
+tutorials/optimization/general_optimization.pot
 tutorials/optimization/using_servers.pot
+tutorials/optimization/cpu_optimization.pot
+tutorials/optimization/gpu_optimization.pot
 tutorials/optimization/using_multimesh.pot
+tutorials/optimization/batching.pot
+tutorials/optimization/optimizing_3d_performance.pot
 tutorials/misc/index.pot
 tutorials/misc/jitter_stutter.pot
 tutorials/misc/handling_quit_requests.pot
@@ -247,6 +269,7 @@ tutorials/misc/instancing_with_signals.pot
 tutorials/misc/state_design_pattern.pot
 tutorials/debug/index.pot
 tutorials/debug/overview_of_debugging_tools.pot
+tutorials/debug/debugger_panel.pot
 tutorials/legal/index.pot
 tutorials/legal/complying_with_licenses.pot
 development/compiling/index.pot
@@ -265,7 +288,16 @@ development/compiling/optimizing_for_size.pot
 development/compiling/compiling_with_script_encryption_key.pot
 development/cpp/index.pot
 development/cpp/introduction_to_godot_development.pot
-development/cpp/configuring_an_ide.pot
+development/cpp/configuring_an_ide/index.pot
+development/cpp/configuring_an_ide/android_studio.pot
+development/cpp/configuring_an_ide/clion.pot
+development/cpp/configuring_an_ide/code_blocks.pot
+development/cpp/configuring_an_ide/kdevelop.pot
+development/cpp/configuring_an_ide/qt_creator.pot
+development/cpp/configuring_an_ide/visual_studio.pot
+development/cpp/configuring_an_ide/visual_studio_code.pot
+development/cpp/configuring_an_ide/xcode.pot
+development/cpp/common_engine_methods_and_macros.pot
 development/cpp/core_types.pot
 development/cpp/variant_class.pot
 development/cpp/object_class.pot
@@ -276,18 +308,21 @@ development/cpp/custom_resource_format_loaders.pot
 development/cpp/custom_audiostreams.pot
 development/cpp/custom_godot_servers.pot
 development/editor/index.pot
-development/editor/introduction.pot
 development/editor/creating_icons.pot
+development/editor/editor_style_guide.pot
 development/file_formats/index.pot
+development/file_formats/gdscript_grammar.pot
 development/file_formats/tscn.pot
 community/contributing/index.pot
 community/contributing/ways_to_contribute.pot
 community/contributing/best_practices_for_engine_contributors.pot
 community/contributing/pr_workflow.pot
+community/contributing/bisecting_regressions.pot
 community/contributing/code_style_guidelines.pot
 community/contributing/bug_triage_guidelines.pot
 community/contributing/documentation_guidelines.pot
 community/contributing/docs_writing_guidelines.pot
 community/contributing/updating_the_class_reference.pot
+community/contributing/editor_and_docs_localization.pot
 community/channels.pot
 community/tutorials.pot
```

And diff since 3.2.3-stable:
```diff
diff --git a/templates_list.txt b/templates_list.txt
index fb598b9e9..f3ae002b9 100644
--- a/templates_list.txt
+++ b/templates_list.txt
@@ -2,6 +2,7 @@ index.pot
 about/index.pot
 about/introduction.pot
 about/faq.pot
+about/troubleshooting.pot
 about/list_of_features.pot
 about/docs_changelog.pot
 about/release_policy.pot
@@ -27,6 +28,7 @@ getting_started/step_by_step/filesystem.pot
 getting_started/step_by_step/scene_tree.pot
 getting_started/step_by_step/singletons_autoload.pot
 getting_started/editor/index.pot
+getting_started/editor/using_the_web_editor.pot
 getting_started/editor/command_line_tutorial.pot
 getting_started/editor/external_editor.pot
 getting_started/editor/default_key_mapping.pot
@@ -141,14 +143,16 @@ tutorials/physics/ray-casting.pot
 tutorials/physics/ragdoll_system.pot
 tutorials/physics/kinematic_character_2d.pot
 tutorials/physics/soft_body.pot
+tutorials/physics/collision_shapes_3d.pot
 tutorials/math/index.pot
 tutorials/math/vector_math.pot
 tutorials/math/vectors_advanced.pot
 tutorials/math/matrices_and_transforms.pot
 tutorials/math/interpolation.pot
 tutorials/math/beziers_and_curves.pot
+tutorials/math/random_number_generation.pot
 tutorials/animation/index.pot
-tutorials/animation/introduction_2d.pot
+tutorials/animation/introduction.pot
 tutorials/animation/cutout_animation.pot
 tutorials/animation/2d_skeletons.pot
 tutorials/animation/animation_tree.pot
@@ -161,7 +165,6 @@ tutorials/io/index.pot
 tutorials/io/background_loading.pot
 tutorials/io/data_paths.pot
 tutorials/io/saving_games.pot
-tutorials/io/encrypting_save_games.pot
 tutorials/i18n/index.pot
 tutorials/i18n/internationalizing_games.pot
 tutorials/i18n/localization_using_gettext.pot
@@ -229,7 +232,9 @@ tutorials/plugins/android/index.pot
 tutorials/plugins/android/android_plugin.pot
 tutorials/platform/index.pot
 tutorials/platform/android_in_app_purchases.pot
-tutorials/platform/services_for_ios.pot
+tutorials/platform/ios/index.pot
+tutorials/platform/ios/ios_plugin.pot
+tutorials/platform/ios/plugins_for_ios.pot
 tutorials/platform/platform_html5.pot
 tutorials/platform/html5_shell_classref.pot
 tutorials/platform/customizing_html5_shell.pot
```

Author: I'm not sure if I understand how to split the list between pages added since 3.2 and pages added since 3.2.3. I think it's enough to split by minor version; I wouldn't bother listing a patch version in the documentation changelog.

Either way, I added some pages I forgot to list thanks to your list :slightly_smiling_face:

Reviewer: My point is that in the pages listed as new since version 3.2, there are many pages which were added *after* 3.2. We cherry-picked them to the `3.2` branch, but they were not in 3.2-stable.

It's not clear to me what we're documented here in the end if we document all changes made between 3.2 and 3.3 as "New pages since version 3.2" and then the 3.3 release has no novelty.
